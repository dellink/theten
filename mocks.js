export const tiles = [
  {
    value: 5,
    row: 0,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'KQ317M0iPCG7dnn5HwBCK',
  },
  {
    value: 6,
    row: 0,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: 'CeITToqKJ4hLt5XzfQuQA',
  },
  {
    value: 6,
    row: 0,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'WI9fDkLCK4bi6aUH4SGi3',
  },
  {
    value: 3,
    row: 0,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'bnYWFOluVTq0v28NIlNVU',
  },
  {
    value: 8,
    row: 1,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'GMqOeXXVnqhxvIG1dONHp',
  },
  {
    value: 5,
    row: 1,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: 'VWONtGPl8u7KwUw3QvOeB',
  },
  {
    value: 1,
    row: 1,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'OqbcyZrr6RvzVHDFr81Nz',
  },
  {
    value: 7,
    row: 1,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'kjxU8k_dTMPzfhga1e52Z',
  },
  {
    value: 7,
    row: 2,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'Hx5pVa3tErUIKZfxGSheo',
  },
  {
    value: 4,
    row: 2,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: '39OGk3pF13ydnZTR53JBR',
  },
  {
    value: 2,
    row: 2,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'xWmRL7Z9GTwNcy8JItY70',
  },
  {
    value: 4,
    row: 2,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'Gz9x2LpfV0YvjpAUpB6pe',
  },
  {
    value: 9,
    row: 3,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'AQZEgW_RVS3BmFxzMkugJ',
  },
  {
    value: 1,
    row: 3,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: 'S5p7c1sYTaUXzKfUYYAf5',
  },
  {
    value: 9,
    row: 3,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'XKnfhJ3YpfNLrM-lDO1Dm',
  },
  {
    value: 2,
    row: 3,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'TLVZ_tlfp-M1BALvwa6AQ',
  },
];
