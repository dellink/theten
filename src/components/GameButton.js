import React from 'react';
import { StyleSheet, Text, TouchableHighlight } from 'react-native';

export default function GameButton({ title, onPress }) {
  return (
    <TouchableHighlight underlayColor="#0288D1" onPress={onPress} style={styles.container}>
      <Text style={styles.text}>{title}</Text>
    </TouchableHighlight>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#01A9F2',
    borderColor: '#0187D0',
    borderRadius: 4,
    borderWidth: 1,
    flex: 1,
    justifyContent: 'center',
    maxHeight: 80,
    width: 140,
  },
  text: {
    alignSelf: 'center',
    color: 'white',
    fontSize: 20,
    fontWeight: '600',
  },
});
