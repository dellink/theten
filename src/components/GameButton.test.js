import React from 'react';
import { render } from '@testing-library/react-native';
import { tiles } from '../../mocks';
import GameButton from './GameButton';

describe(GameButton.name, () => {
  it('should render correctly', () => {
    expect(render(<GameButton tile={tiles[0]} />).asJSON()).toMatchSnapshot();
  });
});
