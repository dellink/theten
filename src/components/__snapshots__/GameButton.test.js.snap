// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameButton should render correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#01A9F2",
        "borderColor": "#0187D0",
        "borderRadius": 4,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "maxHeight": 80,
        "width": 140,
      }
    }
    underlayColor="#0288D1"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 20,
          "fontWeight": "600",
        }
      }
    />
  </>
</View>
`;
